azure-go-sdk-conceptual/azure-sdk-go-authorization.md

```diff
@@ -0,0 +1,137 @@
+
+If you need to connect to a different cloud or management endpoint, you can also set the following environment variables. The most common reasons set them are if you use Azure Stack, a cloud in a different geographic region, or the Azure Classic deployment model.
+
+| Environment variable | Description  | Default |
+|----------------------|--------------|---------|
+| `AZURE_ENVIRONMENT` | The name of the cloud environment to connect to. | `go_autorest.autorest.azure.AzurePublicCloud` |
+| `AZURE_AD_RESOURCE` | The Active Directory resource ID to use when connecting. | `go_autorest.azure.ResourceManagerEndpoint` |
+
+When using environment-based authentication, call the [NewAuthorizerFromEnvironment()](https://godoc.org/github.com/Azure/go-autorest/autorest/azure/auth#NewAuthorizerFromEnvironment) method to get your authorizer object. This object is then set
+on the `Authorizer` property of clients to allow them access to Azure.
+
+```go
+authorizer, err := NewAuthorizerFromEnvironment()
+```
+
+## Use file-based authentication
+
+File-based authentication only works with service principal credentials when they are stored in a local file format generated by [the Azure CLI 2.0](/cli/azure). You can easily create this file when creating a new service principal with the `--sdk-auth` parameter. If you plan on using file-based authentication, make sure that this argument is provided when creating a service principal. Since the CLI prints output to `stdout`, redirect output to a file.
+
+```azurecli
+az ad sp create-for-rbac --sdk-auth > azure.auth
+```
+
+You then set the `AZURE_AUTH_LOCATION` environment variable to where the authorization file is located. This environment variable is read by the application, and the credentials within it are parsed. If you need to select the authorization file at runtime, manipulate the program's environment using the [os.Setenv()](https://golang.org/pkg/os/#Setenv) function.
+
+To load the authentication information, call the [NewAuthorizerFromFile()](https://godoc.org/github.com/Azure/go-autorest/autorest/azure/auth#NewAuthorizerFromFile) function. Unlike environment-based authorization, file-based authorization requires a resource endpoint.
+
+```go
+authorizer, err := NewAuthorizerFromFile(azure.PublicCloud.ResourceManagerEndpoint)
+```
+
+For more on using service principals and managing their access permissions, see [Create a service principal with Azure CLI 2.0].
+
+## Use an authentication client
+
+If you require a specific type of authentication and are willing to have your program do the work to load authentication information from the user, you can use any client that conforms to the [auth.AuthorizerConfig](https://godoc.org/github.com/Azure/go-autorest/autorest/azure/auth#AuthorizerConfig) interface. Use a class that implements this interface when you want an interactive program, specialized configuration files, or have a requirement that prevents you from using another authentication method.

```

Comment:
Minor nit, I would replace the word "class" with "type" since Go doesn't have classes.